x-pack/plugins/alerting/server/routes/lib/rewrite_rule.ts

```diff
@@ -63,10 +63,7 @@ export const rewriteRule = ({
     connector_type_id: actionTypeId,
     ...(frequency
       ? {
-          frequency: {
-            ...frequency,
-            notify_when: frequency.notifyWhen,
-          },
+          frequency,
```

Comment:
why we do not need to do that? I am confused sorry!
https://github.com/elastic/kibana/pull/145637#discussion_r1057874412

Reply:
This fixes a breaking test. Something is broken in the library that converts between camelCase and snake_case inside nested properties. I'll create an issue to investigate further, forgot to do that when I found this

Reply:
https://github.com/elastic/kibana/issues/148572